viewer/src/datamodels/cad/sector/RootSectorNode.ts

```diff
@@ -2,35 +2,36 @@
  * Copyright 2020 Cognite AS
  */
 
+import * as THREE from 'three';
+
 import { SectorNode } from './SectorNode';
 import { CadModelMetadata } from '../CadModelMetadata';
 import { SectorMetadata } from './types';
-import { toThreeMatrix4 } from '@/utilities';
 
 export class RootSectorNode extends SectorNode {
   public readonly sectorNodeMap: Map<number, SectorNode>;
 
   constructor(modelMetadata: CadModelMetadata) {
     super(0, '/');
-    const { scene, modelTransformation } = modelMetadata;
-    this.applyMatrix4(toThreeMatrix4(modelTransformation.modelMatrix));
+    const { scene, modelMatrix } = modelMetadata;
     this.sectorNodeMap = new Map();
     buildScene(scene.root, this, this.sectorNodeMap);
 
     // Disable automatic update of matrices of the subtree as it
     // is quite time consuming. We trust that our owner keeps
     // our matrices updated.
     this.matrixAutoUpdate = false;
+    this.setModelTransformation(modelMatrix);
+  }
+
+  setModelTransformation(matrix: THREE.Matrix4): void {
+    this.matrix.copy(matrix);
     this.updateMatrixWorld(true);
   }
 
-  forceUpdateMatrixWorld(): void {
-    // super.updateMatrixWorld(true);
-    this.traverse(x => {
-      if (x !== this) {
-        x.updateMatrixWorld(true);
-      }
-    });
+  getModelTransformation(out?: THREE.Matrix4): THREE.Matrix4 {
+    out = out ?? new THREE.Matrix4();
+    return out.copy(this.matrix);
```

Comment:
Agree with Joachim here. In general `??` is a bad choice against `||` unless keeping falsy values is something you want to do indeed (not the case here - falsy value will give a runtime error).

Though the main downside of default param is that it will be applied only if undefined is passed. For any other falsy value user will get runtime error too.